diff -Naur mesa-12.0.1.orig/src/egl/main/egl.pc.in mesa-12.0.1/src/egl/main/egl.pc.in
--- mesa-12.0.1.orig/src/egl/main/egl.pc.in	2016-01-18 08:39:25.000000000 +0100
+++ mesa-12.0.1/src/egl/main/egl.pc.in	2016-11-27 20:01:40.284075822 +0100
@@ -7,6 +7,6 @@
 Description: Mesa EGL library
 Requires.private: @GL_PC_REQ_PRIV@
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lEGL
+Libs: -L${libdir}/arm-linux-gnueabihf/tegra-egl -L${libdir} -lEGL
 Libs.private: @GL_PC_LIB_PRIV@
 Cflags: -I${includedir} @GL_PC_CFLAGS@
diff -Naur mesa-12.0.1.orig/src/mapi/es1api/glesv1_cm.pc.in mesa-12.0.1/src/mapi/es1api/glesv1_cm.pc.in
--- mesa-12.0.1.orig/src/mapi/es1api/glesv1_cm.pc.in	2016-01-18 08:39:26.000000000 +0100
+++ mesa-12.0.1/src/mapi/es1api/glesv1_cm.pc.in	2016-11-27 20:04:44.513960821 +0100
@@ -7,6 +7,6 @@
 Description: Mesa OpenGL ES 1.1 CM library
 Requires.private:
 Version: @VERSION@
-Libs: -L${libdir} -lGLESv1_CM
+Libs: -L${libdir}/arm-linux-gnueabihf/tegra-egl -L${libdir} -lGLESv1_CM
 Libs.private: @GLESv1_CM_PC_LIB_PRIV@
 Cflags: -I${includedir}
diff -Naur mesa-12.0.1.orig/src/mapi/es2api/glesv2.pc.in mesa-12.0.1/src/mapi/es2api/glesv2.pc.in
--- mesa-12.0.1.orig/src/mapi/es2api/glesv2.pc.in	2016-01-18 08:39:26.000000000 +0100
+++ mesa-12.0.1/src/mapi/es2api/glesv2.pc.in	2016-11-27 20:04:46.913972352 +0100
@@ -7,6 +7,6 @@
 Description: Mesa OpenGL ES 2.0 library
 Requires.private:
 Version: @VERSION@
-Libs: -L${libdir} -lGLESv2
+Libs: -L${libdir}/arm-linux-gnueabihf/tegra-egl -L${libdir} -lGLESv2
 Libs.private: @GLESv2_PC_LIB_PRIV@
 Cflags: -I${includedir}
diff -Naur mesa-12.0.1.orig/src/mesa/gl.pc.in mesa-12.0.1/src/mesa/gl.pc.in
--- mesa-12.0.1.orig/src/mesa/gl.pc.in	2016-01-18 08:39:26.000000000 +0100
+++ mesa-12.0.1/src/mesa/gl.pc.in	2016-11-27 20:02:52.146421033 +0100
@@ -7,7 +7,7 @@
 Description: Mesa OpenGL library
 Requires.private: @GL_PC_REQ_PRIV@
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -l@GL_LIB@
+Libs: -L${libdir}/arm-linux-gnueabihf/tegra -L${libdir} -l@GL_LIB@
 Libs.private: @GL_PC_LIB_PRIV@
 Cflags: -I${includedir} @GL_PC_CFLAGS@
 glx_tls: @GLX_TLS@
